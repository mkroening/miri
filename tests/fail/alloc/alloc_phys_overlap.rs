extern "Rust" {
    fn miri_alloc_phys(phys_addr: usize, size: usize) -> *mut u8;
}

pub const PHYS_ADDR: usize = 0xB8000;

fn main() {
    unsafe {
        miri_alloc_phys(PHYS_ADDR, 2);
        miri_alloc_phys(PHYS_ADDR + 1, 2); //~ERROR: unsupported operation: trying to allocate physical memory 0xb8001..0xb8003, but 0xb8000..0xb8002 was already allocated
    }
}
