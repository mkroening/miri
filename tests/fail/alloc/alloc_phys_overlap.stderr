error: unsupported operation: trying to allocate physical memory $HEX..$HEX, but $HEX..$HEX was already allocated ($HEX[ALLOC]<TAG>)
  --> $DIR/alloc_phys_overlap.rs:LL:CC
   |
LL |         miri_alloc_phys(PHYS_ADDR + 1, 2);
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ trying to allocate physical memory $HEX..$HEX, but $HEX..$HEX was already allocated ($HEX[ALLOC]<TAG>)
   |
   = help: this is likely not a bug in the program; it indicates that the program performed an operation that the interpreter does not support
   = note: BACKTRACE:
   = note: inside `main` at $DIR/alloc_phys_overlap.rs:LL:CC

note: some details are omitted, run with `MIRIFLAGS=-Zmiri-backtrace=full` for a verbose backtrace

error: aborting due to previous error

